### PR TITLE
[Relay][Topi][CPU] Dense with weight transform

### PR DESCRIPTION
**Pull request by kevinthesun**
This PR adds weight transformed dense to avoid weight layout conversion. With this PR, current dense pack template can achieve 2x performance on CPU targets.

@icemelon9

### CODE REVIEW

File: src/relay/op/nn/nn.h [hunk 1]
```diff
@@ -88,6 +90,29 @@ bool DenseRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
   return true;
 }
 
+template <typename AttrType>
```
Comment (icemelon9) [inline src/relay/op/nn/nn.h:93]: should we keep this function in `nn.cc`?

**Review by icemelon9 — COMMENTED** [review]
Overall LGTM

File: src/relay/op/nn/nn.h (hunk 1 above)
Comment (kevinthesun) [reply src/relay/op/nn/nn.h:93]: I keep ```DensePackRel``` here since ```DenseRel``` is in ```nn.h```.

**Comment by kevinthesun** [comment]
@icemelon9 PTAL

**Review by icemelon9 — APPROVED** [review]
LGTM

**Comment by icemelon9** [comment]
Thanks @kevinthesun

**Pull request merged**